### PR TITLE
slack-bot: Add function to post songs to the users using slack bot

### PR DESCRIPTION
**Pull request by ShishirBhandari**
# Description
Add function to post songs to the user using slack bot.

## Usage
post_slack_message(user_id, video_url, similarity)

### CODE REVIEW

File: musicrs/util/slack_api.py [hunk 1]
```diff
@@ -76,6 +78,28 @@ def retrieve_slack_messages(channel: str, start_date: str, end_date: str):
...
+    type video_id: string
+    """
+    try:
+        dm_response = slackBotClient.conversations_open(users=user_id)
+        dm_channel = dm_response["channel"]["id"]
+        video_url = generate_url(video_id=video_id)
+
+        response = slackBotClient.chat_postMessage(
+            channel=dm_channel,
+            text=video_url
```
Comment (aakritsubedi) [inline musicrs/util/slack_api.py]: we are supposed to include the accuracy percentage also in the text message. 
Reply Message: 'Hello $USER, we have the following song suggestion for you: '
and youtube link  (accuracy: 67%)
- include the accuracy value.

Comment (ShishirBhandari) [reply musicrs/util/slack_api.py]: I don't like this long text,
let's discuss this in our channel.

File: musicrs/util/slack_api.py [hunk 2]
```diff
@@ -69,3 +76,62 @@ def retrieve_slack_messages(channel: str, start_date: str, end_date: str):
...
+        dm_channel = dm_response["channel"]["id"]
+        video_url = generate_url(video_id=video_id)
+
+        response = slackBotClient.chat_postMessage(channel=dm_channel, text=video_url)
+        print("Posted {0} to the user {1}".format(video_url, user_id))
+    except:
+        print("Cannot post video to the user: " + str(user_id))
+
+
+def dump_slack_to_db(slack_message, serialized_np):
```
Comment (Basanta78) [inline musicrs/util/slack_api.py]: This method has moved to service layer. You should have pulled master branch

File: musicrs/util/slack_api.py [hunk 3]
```diff
@@ -69,3 +76,62 @@ def retrieve_slack_messages(channel: str, start_date: str, end_date: str):
...
+                user_name=slack_message["user_real_name"],
+                user_email=slack_message["user_email"],
+                user_profile_picture=slack_message["user_profile_picture"],
+                video_id=video_id,
+                audio_encoding=serialized_np,
+            )
+            session.add(user_profile)
+
+
+def load_slack_messages(start_date, end_date):
```
Comment (Basanta78) [inline musicrs/util/slack_api.py]: This method has moved to service layer. You should have pulled master branch

File: musicrs/util/slack_api.py [hunk 4]
```diff
@@ -69,3 +76,62 @@ def retrieve_slack_messages(channel: str, start_date: str, end_date: str):
...
+    type user_id: string
+    param video_id: youtube video id
+    type video_id: string
+    """
+    try:
+        dm_response = slackBotClient.conversations_open(users=user_id)
+        dm_channel = dm_response["channel"]["id"]
+        video_url = generate_url(video_id=video_id)
+
+        response = slackBotClient.chat_postMessage(channel=dm_channel, text=video_url)
```
Comment (Basanta78) [inline musicrs/util/slack_api.py]: Change message fomat to: You might like this song $url which is $% similar to your interest.

File: musicrs/util/slack_api.py [hunk 5]
```diff
@@ -69,3 +76,62 @@ def retrieve_slack_messages(channel: str, start_date: str, end_date: str):
                 retrieved_messages.append(retrieved_message)
 
     return retrieved_messages
+
+
+def post_slack_message(user_id, video_id):
```
Comment (Basanta78) [inline musicrs/util/slack_api.py]: get similarity as paramter too

File: musicrs/util/slack_api.py [hunk 6]
```diff
@@ -69,3 +76,62 @@ def retrieve_slack_messages(channel: str, start_date: str, end_date: str):
...
+    param video_id: youtube video id
+    type video_id: string
+    """
+    try:
+        dm_response = slackBotClient.conversations_open(users=user_id)
+        dm_channel = dm_response["channel"]["id"]
+        video_url = generate_url(video_id=video_id)
+
+        response = slackBotClient.chat_postMessage(channel=dm_channel, text=video_url)
+        print("Posted {0} to the user {1}".format(video_url, user_id))
```
Comment (Basanta78) [inline musicrs/util/slack_api.py]: Use logger

**Review by Basanta78 — APPROVED** [review]

**Pull request merged**